notebooks/223-text-prediction/223-text-prediction.ipynb

```diff
@@ -0,0 +1,549 @@
+    "        # get next token id\n",
+    "        probs = softmax(next_token_scores)\n",
+    "        next_tokens = np.random.choice(probs.shape[-1], 1,\n",
+    "                                       p=probs[0], replace=True)\n",
+    "        # break the loop if max length or end of text token is reached\n",
+    "        # print(next_tokens[0])\n",
+    "        # print(cur_input_len,max_sequence_length)\n",
+    "        if cur_input_len == max_sequence_length or next_tokens[0] == eos_token_id:\n",
+    "            break\n",
+    "        else:\n",
+    "            input_ids = np.concatenate((input_ids, [next_tokens]), axis=-1)\n",
+    "            attention_mask = np.concatenate((attention_mask, [[1] * len(next_tokens)]), axis=-1)\n",
+    "    return input_ids"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run\n",
+    "The `text` variable below is the input used to generate a predicted sequence."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Generation took 13.061 s\n",
+      "Input Text:  Deep learning is a type of machine learning that uses neural networks\n",
+      "\n",
+      "Predicted Sequence: Deep learning is a type of machine learning that uses neural networks trained from the input data to solve many tasks.\n",
+      "\n",
+      "As of this writing, many recent work has shown significant progress.\n",
+      "\n",
+      "What is more, recent works have also shown that there are several ways that machine learning works. The main one being the use of neural networks in some scenarios, for instance, if humans are using a human, they are able to learn how to learn how to learn from that input data.\n",
+      "\n",
+      "In this section of future work, we aim to describe the neural-network-based approach. We start by explaining that, to be able to\n"
+     ]
+    }
+   ],
+   "source": [
+    "import time\n",
+    "if not conversational:\n",
+    "    text = \"Deep learning is a type of machine learning that uses neural networks\"\n",
+    "    input_ids, attention_mask = tokenize(text)\n",
+    "\n",
+    "    start = time.perf_counter()\n",
+    "    output_ids = generate_sequence(input_ids, attention_mask)\n",
+    "    end = time.perf_counter()\n",
+    "    output_text = \" \"\n",
+    "    # Convert IDs to words and make the sentence from it\n",
+    "    for i in output_ids[0]:\n",
+    "        output_text += tokenizer.batch_decode([i])[0]\n",
+    "    print(f\"Generation took {end - start:.3f} s\")\n",
+    "    print(f\"Input Text:  {text}\")\n",
+    "    print()\n",
+    "    print(f\"Predicted Sequence:{output_text}\")\n",
+    "else:\n",
+    "    print(\"Notebook Mode is Conversationsational, set conversational = False in the first cell\")"
```

Comment:
Request to rephrase the print message.
- Notebook mode is set to conversational. Please set conversational = False in the first cell to generate text sequences.